### PR TITLE
k6-operator added for load testing

### PR DESCRIPTION
**Pull request by logoutdhaval**
## Description

* Describe the changes made and why they were made.
* Add a link to teh design document or include the design bullet points related to this PR here.

 _(Ignore if these details are present on the associated JIRA ticket)_

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Design related bullet points or design document link related to this PR added in the description above. 

- [ ] Updated corresponding Postman Collection or Api documentation for the changes in this PR.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Add required Swagger annotation and update API documentation with details of any API changes if applicable

- [ ] Followed the naming conventions as given in https://docs.google.com/document/d/1Q4vaMSzrTxxh9TS0RILuNkSkYCxotuYk1Xe0CMIkkCU/edit?usp=sharing

### CODE REVIEW

File: helm/k6-operator/templates/prometheus/serviceMonitor.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{{- if .Values.prometheus.enabled }}
+# Prometheus Monitor Service (Metrics)
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
```
Comment (somanath21) [inline helm/k6-operator/templates/prometheus/serviceMonitor.yaml:4]: Prometheus has well known helm charts. Adding K8s manifests for these are not maintainable (unnecessary technical debt). Please replace with helm chart overriding of values.yaml if required.

File: helm/k6-operator/e2e/test.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+//postman[Symbol.for("initial")]({
+//  options,
+//});
+//
+//export default function () {
+//  postman[Request]({
+//    name: "message-gateway sms API",
+//    id: "0c5834c9-468c-4a33-961d-f1e2278ac61a",
+//    method: "POST",
+//    address: "https://messagegateway.sandbox.fynarfin.io/sms",
```
Comment (somanath21) [inline helm/k6-operator/e2e/test.js:35]: Do you need this dead code?

File: helm/k6-operator/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+apiVersion: v1
+appVersion: "0.0.6"
```
Comment (somanath21) [inline helm/k6-operator/Chart.yaml:2]: Don't duplicate code unnecessarily. Use this helm chart as a dependency and only override fields you require in values.yaml.

File: helm/k6-operator/e2e/test.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+//}/Users/dhavalmaniyar/Documents/projects/k6-helm-chart/k6-operator/e2e/test.js
+import http from 'k6/http';
+export const options = {
+  insecureSkipTLSVerify: true,
+  vus: 5,
+  iterations: 10,
+};
+export default function () {
+  const url = 'https://messagegateway.sandbox.fynarfin.io/sms';
+  const payload = JSON.stringify([{
```
Comment (somanath21) [inline helm/k6-operator/e2e/test.js:53]: This part looks like the only code other than values / Chart.yaml that might be required for this PR.

File: helm/k6-operator/e2e/test.yaml [hunk 5]
```diff
@@ -0,0 +1,11 @@
+---
+apiVersion: k6.io/v1alpha1
+kind: K6
+metadata:
+  name: k6-sample
+spec:
```
Comment (somanath21) [inline helm/k6-operator/e2e/test.yaml:6]: You can let this remain in the PR.

**Review by somanath21 — CHANGES_REQUESTED** [review]
97% of the code looks unnecessarily duplicated code.  Use this helm chart as a dependency and only override fields you require in values.yaml.

**Comment by fynmanoj** [comment]
@logoutdhaval please address comments and get this PR merged

**Pull request closed without merging**